pkg/runctl/run_manager.go

```diff
@@ -457,7 +461,14 @@ func (c *runManager) getServiceAccountSecretName(ctx *runContext) string {
 		return c.testing.getServiceAccountSecretNameStub(ctx)
 	}
 
-	return ctx.serviceAccount.GetHelper().GetServiceAccountSecretNameRepeat()
+	start := time.Now()
+	secretName := ctx.serviceAccount.GetHelper().GetServiceAccountSecretNameRepeat()
+	end := time.Now()
+	elapsed := end.Sub(start)
+	klog.V(6).Infof("getServiceAccountSecretName for %q took %v", ctx.pipelineRun.String(), elapsed)
+
+	c.metrics.ObserveRetryDurationByType("ServiceAccountSecret", elapsed)
```

Comment:
Can be be sure here that it is a retry? In the other case (`CommitStatus()` there is a boolean flag indicating a real retry and we call `ObserveRetryDurationByType` only in case of a real retry.  Here we don't know if a re-try happend when we call `GetServiceAccountSecretNameRepeat()`. It might succeed with the first attempt.

Reply:
In the commit case we most likely are successful with the first try so I decided to only meter in the error case. A retry always indicates an error.

In the account secret case we are waiting until the secret exists and retrying is much more likely than in the first case. A retrying does not indicate an error, just a slow processing. I decided to don't distinguish  if  we have real retrying or if it just succedes iwth the first run.

Reply:
The type name `ServiceAccountSecret` is not self-explanatory. You have to know the implementation to understand what is meant. Assuming we use this metric for _many_ code places with retries it becomes hard to give each of them a good name.

Instead of inventing names for the retry location, we could simply use the program structure itself. The observation function could obtain information about its caller and use it for labeling. One label could specify only the direct caller, while another label could include the top N frames of the call stack. See [`runtime.Callers()`](https://pkg.go.dev/runtime#Callers) and [`runtime.CallFrames()`](https://pkg.go.dev/runtime#CallersFrames) for details.